woocommerce/woocommerce-android
# Nick/simple payments cardreader

**This is being submitted as a draft** because I'm still experimenting with a better approach.

This PR addresses taking a card payment after the simple payment flow. Because the card reader flow is tightly coupled to the order detail and the settings flows, this left me with three options:

1. Rewrite the card reader flow so it could be more easily shared from multiple navigation graphs
2. Duplicate the card reader flow and adapt it for simple payments
3. Have the simple payments flow show order detail and tell it to launch the card reader flow

The first approach is the cleanest, but after spending a couple days on it I realized how big a task it is.  We'd have to rewrite and re-test much of the current card reader flow in settings and order detail. Given our timeline, I ruled it out.

The second was a possibility, and it may be doable in our time frame. but the idea of having three separate card reader flows didn't sit well with me.

The last option isn't the most user friendly, and it feels almost hacky, but it has the benefit of being by far the quickest to implement so I opted for it. We can revisit the first two options later.

To test:

* Go to the order list and create a simple payment
* Tap Take Payment
* Choose Card payment
* Verify that the user is taken to order detail and the card payment flow is started
